### PR TITLE
UUID changes

### PR DESCRIPTION
**Pull request by crimsonred**
refactor(uuid): BREAKING CHANGES, UUID is not automatically generated by the SDK anymore. If NOT set a panic will occur.

UUID is not automatically generated by the SDK anymore. Please set it in the PNConfig before instantiating the PubNub object. If UUID is empty panic will occur.

BREAKING CHANGES: UUID is not automatically generated by the SDK anymore. If NOT set a panic will occur.

### CODE REVIEW

File: examples/cli/cli_demo.go [hunk 1]
```diff
@@ -1471,14 +1440,14 @@ func granttoken(args []string) {
...
-		channelMetadatas = strings.Split(args[3], ",")
-	}
+	// var users []string
+	// if len(args) > 2 {
+	// 	users = strings.Split(args[2], ",")
+	// }
+	// var channelMetadatas []string
+	// if len(args) > 3 {
+	// 	channelMetadatas = strings.Split(args[3], ",")
+	// }
```
Comment (kleewho) [inline examples/cli/cli_demo.go:1450]: IMHO all commented out lines can stay removed

**Review by kleewho — COMMENTED** [review]
I wonder why not put explicit requirement in NewConfig `func NewConfig(uuid string) *Config`?

**Comment by crimsonred** [comment]
> I wonder why not put explicit requirement in NewConfig func NewConfig(uuid string) *Config?

Not in spec

**Comment by crimsonred** [comment]
@client-engineering-bot release

**Comment by kleewho** [comment]
> > I wonder why not put explicit requirement in NewConfig func NewConfig(uuid string) *Config?
> 
> Not in spec

The "spec" says:

> We should validate uuid “as soon as we can” - string can’t be empty
For compiled languages use type system to validate uuid is set during compile time

What we have here right now is not a type system compile time change

**Comment by crimsonred** [comment]
> The "spec" says:

> We should validate uuid “as soon as we can” - string can’t be empty
For compiled languages use type system to validate uuid is set during compile time

@kleewho  The ticket for Go or the Epic doesn't say anything like that

**Comment by crimsonred** [comment]
@client-engineering-bot release

File: config.go [hunk 2]
```diff
@@ -50,19 +48,21 @@ type Config struct {
 
 // NewDemoConfig initiates the config with demo keys, for tests only.
 func NewDemoConfig() *Config {
-	demoConfig := NewConfig()
+	demoConfig := NewConfig(GenerateUUID())
 
 	demoConfig.PublishKey = "demo"
 	demoConfig.SubscribeKey = "demo"
 	demoConfig.SecretKey = "demo"
+	demoConfig.UUID = GenerateUUID()
```
Comment (are) [inline config.go:56]: Question (I don't know Go :P) - why does it seem like the UUID is set twice? Once in the constructor and once here?

**Review by are — DISMISSED** [review]

File: config.go (hunk 2 above)
Comment (kleewho) [reply config.go:56]: That's my bad. Forgot to remove it

**Review by are — DISMISSED** [review]

**Review by are — APPROVED** [review]

**Review by kleewho — APPROVED** [review]

**Pull request merged**